.circleci/config.yml

```diff
@@ -185,6 +184,32 @@ jobs:
           name: 'Initialize Workspace'
           command: cp .circleci/maybe_gracefully_halt.sh /tmp/workspace
       - teardown_vm
+  initialize_node:
+    executor:
+      name: node-docker-large
+    steps:
+      - run:
+          name: 'Fetch package.json and required files'
+          command: |
+            mkdir -p build-system/{common,task-runner}
+            wget https://raw.githubusercontent.com/${CIRCLE_PROJECT_USERNAME}/${CIRCLE_PROJECT_REPONAME}/${CIRCLE_SHA1}/package{,-lock}.json
+            wget https://raw.githubusercontent.com/${CIRCLE_PROJECT_USERNAME}/${CIRCLE_PROJECT_REPONAME}/${CIRCLE_SHA1}/build-system/common/{check-package-manager,ci,logging,process}.js --directory-prefix build-system/common
+            wget https://raw.githubusercontent.com/${CIRCLE_PROJECT_USERNAME}/${CIRCLE_PROJECT_REPONAME}/${CIRCLE_SHA1}/build-system/task-runner/{amp-cli-runner,install-amp-task-runner}.js --directory-prefix build-system/task-runner
```

Comment:
Disapprove because this is more complicated than it is worth IMO

Reply:
That's a fair critique :)
I think it might still be worth performing this micro-optimization if we can think of a way to simplify this step?

Reply:
+1 to Esther's comment. I'm not convinced that the complexity and additional maintenance burden introduced by this PR is worth the potential savings of time. In addition, I'm concerned that storing / copying `node_modules/` instead of doing a clean install using `npm ci` could result in hard to debug errors in future.

A less risky alternative IMO would be to persist the `.npm` cache directory so that packages aren't freshly downloaded by each job, and continue to freshly create `node_modules/` by running `npm ci`. We tried this a while ago via #32429 and #32439 without success, not sure if anything has changed.

Reply:
`.npm` is being cached. The issue we had in the past with caching `node_modules/` was that it wasn't always hitting the correct cache - in this case the correct directory is _always_ present because it gets build from the same exact `package.json` file. Maybe I can try the caching route again while paying attention to the caching subtleties now that we know more